tests/pages/multipleOperatorsServiceList.test.tsx

```diff
@@ -0,0 +1,252 @@
+                            ],
+                        },
+                        [MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE]: [
+                            {
+                                nocCode: 'N1',
+                                services: ['service one', 'service two'],
+                            },
+                        ],
+                    },
+                });
+                const result = await getServerSideProps(ctx);
+                const expectedCheckedServiceList: ServicesInfo[] = mockServices.map(mockService => ({
+                    ...mockService,
+                    checked: false,
+                }));
+                expect(result.props.errors.length).toBe(0);
+                expect(result.props.serviceList).toEqual(expectedCheckedServiceList);
+                expect(result.props.buttonText).toEqual('Select All Services');
+                expect(result.props.operatorName).toBe('Test2' || 'Test3');
+                expect(result.props.nocCode).toBe('N2' || 'N3');
+            });
+            it('should return expected props to the page when the page is visited by the user for the third time', async () => {
+                const ctx = getMockContext({
+                    session: {
+                        [MULTIPLE_OPERATOR_ATTRIBUTE]: {
+                            selectedOperators: [
+                                { operatorPublicName: 'Test1', nocCode: 'N1' },
+                                { operatorPublicName: 'Test2', nocCode: 'N2' },
+                                { operatorPublicName: 'Test3', nocCode: 'N3' },
+                            ],
+                        },
+                        [MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE]: [
+                            {
+                                nocCode: 'N1',
+                                services: ['service one', 'service two'],
+                            },
+                            {
+                                nocCode: 'N2',
+                                services: ['service one', 'service two'],
+                            },
+                        ],
+                    },
+                });
+                const result = await getServerSideProps(ctx);
+                const expectedCheckedServiceList: ServicesInfo[] = mockServices.map(mockService => ({
+                    ...mockService,
+                    checked: false,
+                }));
+                expect(result.props.errors.length).toBe(0);
+                expect(result.props.serviceList).toEqual(expectedCheckedServiceList);
+                expect(result.props.buttonText).toEqual('Select All Services');
+                expect(result.props.operatorName).toBe('Test3');
+                expect(result.props.nocCode).toBe('N3');
+            });
+            it('should return expected props to the page when the page is visited by the user for the fourth time having finished all their operators', async () => {
```

Comment:
Is this test checking what happens when the user presses the back button in the browser?

Reply:
Sort of, yeah.